soc/arm/rpi_pico/rp2/soc.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2016 Linaro Limited
+ * Copyright (c) 2021 Yonatan Schachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file SoC configuration macros for the Raspberry Pi RP2040 family processors
+ */
+
+#ifndef _RPI_PICO_RP2040_SOC_H_
+#define _RPI_PICO_RP2040_SOC_H_
+
+#define __VTOR_PRESENT 1
+#define __MPU_PRESENT 1
```

Comment:
where are these used?

Reply:
`__VTOR_PRESENT` because otherwise:
```
zephyr/arch/arm/core/aarch32/prep_c.c: In function 'relocate_vector_table':
zephyr/arch/arm/core/aarch32/prep_c.c:51:5: error: 'SCB_Type' has no member named 'VTOR'
   51 |  SCB->VTOR = VECTOR_ADDRESS & SCB_VTOR_TBLOFF_Msk;
      |     ^~
zephyr/arch/arm/core/aarch32/prep_c.c:51:31: error: 'SCB_VTOR_TBLOFF_Msk' undeclared (first use in this function)
   51 |  SCB->VTOR = VECTOR_ADDRESS & SCB_VTOR_TBLOFF_Msk;
      |                               ^~~~~~~~~~~~~~~~~~~
```

`__MPU_PRESENT` because otherwise:
```
zephyr/arch/arm/core/aarch32/cortex_m/scb.c: In function 'z_arm_clear_arm_mpu_config':
zephyr/arch/arm/core/aarch32/cortex_m/scb.c:57:5: error: 'MPU' undeclared (first use in this function)
   57 |   ((MPU->TYPE & MPU_TYPE_DREGION_Msk) >> MPU_TYPE_DREGION_Pos);
      |     ^~~
[this goes on]
```

Reply:
Putting it in the HAL's autogen doesn't work, because these definitions are needed by `zephyr/arch/arm/core/aarch32/cortex_m/scb.c`, which does not include autogen. As far as I know, `soc.h` is the only reasonable place to put it. But I will change the ones to the correct configs

Reply:
Right, so it's really Zephyr code that is expecting this to come via `soc.h`. I guess it can be left as it is then.